### PR TITLE
User can signup

### PR DESCRIPTION
**Pull request by DevJosephLee**
https://user-images.githubusercontent.com/68756038/147692772-e033d5f4-6011-4176-a08f-6dd207cf3ab7.mp4

Screen Shot 2021-12-29 at 10 28 37 AM

### CODE REVIEW

File: client/pages/sign-in.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(newUser)
+    })
+      .then(response => response.json())
+      .then(newUser => {
+        this.setState({ users: [].concat(this.state.users, newUser) });
+      })
+      .catch(err => console.error(err));
```
Comment (thebearingedge) [inline client/pages/sign-in.jsx:34]: Once the user is signed up, you can switch them to the sign in page (which can be blank for now)

```js
window.location.hash = 'sign-in'
```

File: server/index.js [hunk 2]
```diff
@@ -92,6 +94,47 @@ app.get('/api/reviews/:chefId', (req, res) => {
...
+    .then(result => {
+      const [users] = result.rows;
+      if (!users) {
+        res.status(404).json({ error: 'cannot find any users' });
+      } else {
+        res.json(result.rows);
+      }
+    })
+    .catch(err => console.error(err));
+});
```
Comment (thebearingedge) [inline server/index.js:112]: We generally don't load all of the app's users at any given time.

File: client/pages/sign-in.jsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  }
+
+  componentDidMount() {
+    fetch('/api/users')
+      .then(response => response.json())
+      .then(data => {
+        this.setState({ users: data });
+      })
+      .catch(err => console.error(err));
+  }
```
Comment (thebearingedge) [inline client/pages/sign-in.jsx:20]: I don't recommend fetching and storing the users in the front end.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Comment by DevJosephLee** [comment]
https://user-images.githubusercontent.com/68756038/147698994-4b26ac68-adcb-44d3-a71b-dde36b4940a9.mp4

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**